dhis2/src/main/java/org/motechproject/dhis2/rest/service/impl/DhisWebServiceImpl.java

```diff
@@ -547,17 +524,27 @@ private DhisDataValueStatusResponse importDataValues(Settings settings, String u
         HttpUriRequest request = generatePostRequest(settings, uri, json);
         LOGGER.debug(String.format("Initiating request to create resource: %s, request: %s", json, request.toString()));
         CloseableHttpResponse response = getResponseForRequest(request);
-        LOGGER.debug(String.format("Received response to create resource: %s, request: %s", json, response));
+        return (DhisDataValueStatusResponse) getDhisResponseForHttpResponse(json, uri, response, DhisDataValueStatusResponse.class);
+    }
 
+    private Object getDhisResponseForHttpResponse(String json, String uri, CloseableHttpResponse response, Class responseType) {
+        Object status;
         try (InputStream content = getContentForResponse(response)) {
-            return new ObjectMapper().readValue(content, DhisDataValueStatusResponse.class);
+            String contentString = IOUtils.toString(content);
+            LOGGER.debug(String.format("Received response to create resource: %s, request: %s", contentString, json));
+            status = new ObjectMapper().readValue(contentString, responseType);
+            if (status.toString().contains("ERROR")) {
```

Comment:
you should use enum here instead of string. I think that the best solution will be creating new AbstractClass. DhisDataValueStatusResponse and DhisStatusResponse should extend this new class and then you can use here getStatus method